lib/identity_cache/cached/reference/belongs_to.rb

```diff
@@ -0,0 +1,46 @@
+module IdentityCache
+  module Cached
+    module Reference
+      class BelongsTo < Association # :nodoc:
+        def initialize(name, reflection:)
+          super(name, inverse_name: nil, reflection: reflection)
+        end
+
+        def build
+          reflection.active_record.class_eval(<<-RUBY, __FILE__, __LINE__ + 1)
+            def #{cached_accessor_name}
+              association_klass = association(:#{name}).klass
+              if association_klass.should_use_cache? && #{reflection.foreign_key}.present? && !association(:#{name}).loaded?
+                if instance_variable_defined?(:#{records_variable_name})
```

Comment:
We don't need to use generic instance variable methods with a literal

```suggestion
                if defined?(#{records_variable_name})
```